nodejs/week3/meal_sharing_website/src/backend/api/meals.js

```diff
@@ -0,0 +1,184 @@
+const express = require("express");
+const app = express();
+const router = express.Router();
+const moment = require('moment');
+const pool = require("./../database");
+
+// This function will be invoked when endpoint is passed with params
+const mealsFunctionParams = function (request, response) {
+  const idString = request.params.id;
+  const id = parseInt(idString);
+  if (isNaN(id)) {
+    response.send('Id should be an integer');
+  } else {
+    pool.query('SELECT * FROM meal where id = ?', idString, function (error, results) {
+      if (error) {
+        console.error(`error ${error}`);
+      } else {
+        processArrayLength(results, response);
+      }
+    })
+  }
+}
+
+// This function will be invoked when the endpoint has no params and has query parameters
+const mealsFunctionQuery = function (request, response) {
+  let checkCondition = isValidQueryParam(request.query);
+  if (checkCondition !== null) {
+    response.send(checkCondition);
+  } else {
+    let query = getMealsDBquery(request.query);
+    pool.query(query, function (error, results) {
+      if (error) {
+        console.error(`error ${error}`);
+      }else if (results.length === 0){
+        response.send('No Match Found');
+      } else {
+        response.json(results);
+      }
+    })
+  }
+};
+
+// This function is for POST operation
+const mealsFunctionPost = function (request, response) {
+  //const guestcount = request.body.guestcount;
+  //const mealid = request.body.mealid;
+  const { title, location, when, max_reservations, price } = request.body;
+  const data = {
+    title: title,
+    location: location,
+    when: when,
+    max_reservations: max_reservations,
+    price: price
+  }
+  console.log(data);
+
+  if (!title || !location || !when || !max_reservations) {
```

Comment:
That's a bit odd to have this check *after* you already created an object `data` 🙂 Not a big deal though.